flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/MiniDispatcherTest.java

```diff
@@ -0,0 +1,208 @@
+
+	private static TestingRpcService rpcService;
+
+	private static Configuration configuration;
+
+	private static BlobServer blobServer;
+
+	private MiniDispatcher miniDispatcher;
+
+	private CompletableFuture<JobGraph> jobGraphFuture;
+
+	private TestingLeaderElectionService dispatcherLeaderElectionService;
+
+	@BeforeClass
+	public static void setupClass() throws IOException {
+		rpcService = new TestingRpcService();
+		configuration = new Configuration();
+
+		configuration.setString(BlobServerOptions.STORAGE_DIRECTORY, temporaryFolder.newFolder().getAbsolutePath());
+
+		blobServer = new BlobServer(configuration, new VoidBlobStore());
+	}
+
+	@Before
+	public void setup() throws Exception {
+		dispatcherLeaderElectionService = new TestingLeaderElectionService();
+		final TestingHighAvailabilityServices highAvailabilityServices = new TestingHighAvailabilityServices();
+		final TestingResourceManagerGateway resourceManagerGateway = new TestingResourceManagerGateway();
+		final HeartbeatServices heartbeatServices = new HeartbeatServices(1000L, 1000L);
+		final ArchivedExecutionGraphStore archivedExecutionGraphStore = new MemoryArchivedExecutionGraphStore();
+		final TestingFatalErrorHandler testingFatalErrorHandler = new TestingFatalErrorHandler();
+
+		highAvailabilityServices.setDispatcherLeaderElectionService(dispatcherLeaderElectionService);
+
+		jobGraphFuture = new CompletableFuture<>();
+		final TestingJobManagerRunnerFactory testingJobManagerRunnerFactory = new TestingJobManagerRunnerFactory(jobGraphFuture);
+
+		miniDispatcher = new MiniDispatcher(
+			rpcService,
+			UUID.randomUUID().toString(),
+			configuration,
+			highAvailabilityServices,
+			resourceManagerGateway,
+			blobServer,
+			heartbeatServices,
+			NoOpMetricRegistry.INSTANCE,
+			archivedExecutionGraphStore,
+			testingJobManagerRunnerFactory,
+			testingFatalErrorHandler,
+			null,
+			jobGraph,
+			ClusterEntrypoint.ExecutionMode.DETACHED);
+
+		miniDispatcher.start();
+	}
+
+	@After
+	public void teardown() throws InterruptedException, ExecutionException, TimeoutException {
+		if (miniDispatcher != null) {
+			RpcUtils.terminateRpcEndpoint(miniDispatcher, timeout);
+			miniDispatcher = null;
+		}
+	}
+
+	@AfterClass
+	public static void teardownClass() throws IOException {
+		blobServer.close();
+		rpcService.stopService();
+	}
+
+	/**
+	 * Tests that the {@link MiniDispatcher} recovers the single job with which it
+	 * was started.
+	 */
+	@Test
+	public void testSingleJobRecovery() throws Exception {
+		// wait until the Dispatcher is the leader
+		dispatcherLeaderElectionService.isLeader(UUID.randomUUID()).get();
+
+		final JobGraph actualJobGraph = jobGraphFuture.get();
+
+		assertThat(actualJobGraph.getJobID(), is(jobGraph.getJobID()));
+	}
+
+	/**
+	 * Tests that in detached mode, the {@link MiniDispatcher} will terminate after the job
+	 * has completed.
+	 */
+	@Test
+	public void testTerminationAfterJobCompletion() throws Exception {
```

Comment:
It seems that we are only testing the absence of exceptions here. There are assertions.

Reply:
Will add one.

Reply:
Well we test that the `MiniDispatcher` terminates without throwing an exception. This condition is true if the termination future is completed without an exception. That's what happens when calling `get()`.

Reply:
Ok. I thought the future can be completed with a boolean but it seems to be always completed with `null`.